### PR TITLE
[#9382] Factor out components and add disabled version to Instructor help page, students section

### PR DESCRIPTION
**Pull request by tran-tien-dat**
Part of #9382 

Add a component called `tm-example-box` to wrap around the disabled component.

The disabled component is written as a separate class in the same file, and it inherits from an interface that contains identifiers that are referenced by the shared template.

### CODE REVIEW

File: src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.spec.ts [hunk 1]
```diff
@@ -1,21 +1,25 @@
 import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { Component } from '@angular/core';
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
-import { FormsModule, ReactiveFormsModule } from '@angular/forms';
 import { RouterTestingModule } from '@angular/router/testing';
 import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
 import { InstructorCoursesPageComponent } from './instructor-courses-page.component';
 
+@Component({ selector: 'tm-add-course-form', template: '' })
+class AddCourseFormStubComponent {}
```
[the remark this answers is not in the document]
Comment (tran-tien-dat) [reply src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.spec.ts]: Many times, the components that need to stubbed are in the same module as the component to be tested. So if I import that module to the test module, the component to be tested would be declared in 2 modules. So I just stub everything.

File: src/web/app/pages-instructor/student-profile/course-related-info/course-related-info.component.html [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        <div class="col-sm-12">
+          <h4 class="font-weight-bold">Enrollment Details</h4>
+        </div>
+      </div>
+      <div class="form-group student-info-row row">
+        <div class="col-4 text-right">
+          <label>Course</label>
+        </div>
+        <div class="col-8">
+          <a routerLink="/web/instructor/courses/details" [queryParams]="{courseid: student?.course}">
```
[the remark this answers is not in the document]
Comment (tran-tien-dat) [reply src/web/app/pages-instructor/student-profile/course-related-info/course-related-info.component.html]: I was lazy to do this for so many components. So what I did was using `canDeactivate` guard to disable router navigation in the help page, except for a few whitelisted links on the navigation bar

https://github.com/TEAMMATES/teammates/blob/28768a1b0d7b7839d8ced1b199e55d2bd7c36e0f/src/web/app/pages-instructor/instructor-pages.module.ts#L174-L176

https://github.com/TEAMMATES/teammates/blob/28768a1b0d7b7839d8ced1b199e55d2bd7c36e0f/src/web/app/pages-instructor/instructor-pages.module.ts#L223-L226

File: src/web/app/pages-instructor/instructor-search-page/student-result-table/student-result-table.component.scss [hunk 3]
```diff
@@ -0,0 +1,11 @@
+.bg-light-blue {
+  background-color: #007BFF33 !important;
```
[the remark this answers is not in the document]
Comment (tran-tien-dat) [reply src/web/app/pages-instructor/instructor-search-page/student-result-table/student-result-table.component.scss]: Yes. It is RGBA so the last value is alpha.

File: src/web/app/pages-help/instructor-help-page/example-box/example-box.component.scss [hunk 4]
```diff
@@ -0,0 +1,62 @@
+.bs-example {
+  background-color: #FAFAFA;
+  border-color: #E5E5E5 #EEE #EEE;
+  border-style: solid;
+  border-width: 1px 0;
+  -moz-box-shadow: inset 0 3px 6px rgba(0, 0, 0, .05);
+  -webkit-box-shadow: inset 0 3px 6px rgba(0, 0, 0, .05);
```
Comment (wkurniawan07) [inline src/web/app/pages-help/instructor-help-page/example-box/example-box.component.scss]: You shouldn't need this. Angular CLI has autoprefixer built-in. Similarly for other vendor-prefixed style items.

File: src/web/app/pages-instructor/student-profile/course-related-info/course-related-info.component.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (wkurniawan07) [reply src/web/app/pages-instructor/student-profile/course-related-info/course-related-info.component.html]: Alright, that works, too.

**Pull request merged**